mgc/sdk/static/object_storage/common/path_utils.go

```diff
@@ -0,0 +1,46 @@
+package common
+
+import (
+	"net/url"
+	"strings"
+)
+
+// NormalizeFilePath normalizes file paths for consistent handling across different systems
+// It decodes URL-encoded characters, normalizes directory separators, and removes various prefixes
+func NormalizeFilePath(srcPath string) string {
+	decodedPath, err := url.PathUnescape(srcPath)
+	if err != nil {
+		decodedPath = srcPath
+	}	
+	normalizedPath := strings.ReplaceAll(decodedPath, "\\", "/")
+	normalizedPath = strings.TrimPrefix(normalizedPath, "path://")
+	normalizedPath = strings.TrimPrefix(normalizedPath, "./")
+	normalizedPath = strings.TrimPrefix(normalizedPath, ".")
+	
+	return normalizedPath
+}
+
+// ExtractFileName extracts just the filename from a path
+func ExtractFileName(path string) string {
+	normalizedPath := NormalizeFilePath(path)
+
+	var fileName string
+	if lastSlash := strings.LastIndex(normalizedPath, "/"); lastSlash >= 0 {
+		fileName = normalizedPath[lastSlash+1:]
```

Comment:
Para a string "pasta/", isso retorna vazio
Existe a possibilidade de index out of bounds.
Pode utilizar https://pkg.go.dev/strings#Split ou https://pkg.go.dev/path/filepath#Base